kernel/src/cpu/idt/svsm.rs

```diff
@@ -94,6 +99,20 @@ pub fn early_idt_init() {
 pub fn idt_init() {
     // Set IST vectors
     init_ist_vectors();
+
+    // Capture an address that can be used by assembly code to read the #HV
+    // doorbell page.  The address of each CPU's doorbell page may be
+    // different, but the address of the field in the PerCpu structure that
+    // holds the actual pointer is constant across all CPUs, so that is the
+    // pointer that is actually captured.  The address that is captured is
+    // stored as a usize instead of a typed value, because the declarations
+    // required for type safety here are cumbersome, and the assembly code
+    // that uses the value is not type safe in any case, so enforcing type
+    // safety on the pointer would offer no meaningful value.
+    unsafe {
+        let global_ref = ptr::addr_of!(HV_DOORBELL_ADDR) as usize;
+        *(global_ref as *mut usize) = (*this_cpu_unsafe()).hv_doorbell_addr();
+    }
```

Comment:
We can just do this if we set `HV_DOORBELL_ADDR` to be `static mut`.

```suggestion
    unsafe {
        let global_ref = ptr::addr_of_mut!(HV_DOORBELL_ADDR);
        *global_ref = (*this_cpu_unsafe()).hv_doorbell_addr();
    }
```